src/screens/TaskDetailScreen.tsx

```diff
@@ -191,255 +149,214 @@ const TaskDetailScreen: React.FC<TaskDetailScreenProps> = ({ navigation, route }
+                styles.statusText,
+                task.status === 'completed' ? styles.completedText : styles.pendingText
+              ]}>
+                {task.status === 'completed' ? 'Completed' : 'Pending'}
               </Text>
-            </TouchableOpacity>
-
-            {/* Share button with icon */}
-            <TouchableOpacity
-                style={[styles.button, styles.shareButton]}
-                onPress={handleShare}
-            >
-              <Ionicons name="share-outline" size={18} color="#fff" style={styles.buttonIcon} />
-              <Text style={styles.buttonText}>Share</Text>
-            </TouchableOpacity>
-
-            {/* Edit button with icon */}
-            <TouchableOpacity
-                style={[styles.button, styles.editButton]}
-                onPress={handleEdit}
-            >
-              <Ionicons name="create-outline" size={18} color="#fff" style={styles.buttonIcon} />
-              <Text style={styles.buttonText}>Edit</Text>
-            </TouchableOpacity>
-
-            {/* Delete button with icon */}
-            <TouchableOpacity
-                style={[styles.button, styles.deleteButton]}
-                onPress={handleDelete}
-            >
-              <Ionicons name="trash-outline" size={18} color="#fff" style={styles.buttonIcon} />
-              <Text style={styles.buttonText}>Delete</Text>
-            </TouchableOpacity>
+            </View>
           </View>
-        </Animated.ScrollView>
-      </>
+        </View>
+
+        {/* Action buttons */}
+        <View style={styles.buttonsContainer}>
+          {/* Dynamic toggle status button */}
+          <TouchableOpacity
+              style={[
+                styles.button,
+                task.status === 'completed' ? styles.pendingButton : styles.completeButton
+              ]}
+              onPress={handleToggleStatus}
+          >
+            <Text style={styles.buttonText}>
+              {task.status === 'completed' ? 'Mark as Pending' : 'Mark as Completed'}
+            </Text>
+          </TouchableOpacity>
+
+          {/* Share button */}
+          <TouchableOpacity
+              style={[styles.button, styles.shareButton]}
+              onPress={handleShare}
+          >
+            <Text style={styles.buttonText}>Share</Text>
+          </TouchableOpacity>
+
+          {/* Edit button */}
+          <TouchableOpacity
+              style={[styles.button, styles.editButton]}
+              onPress={handleEdit}
+          >
+            <Text style={styles.buttonText}>Edit</Text>
+          </TouchableOpacity>
+
+          {/* Delete button */}
+          <TouchableOpacity
+              style={[styles.button, styles.deleteButton]}
+              onPress={handleDelete}
+          >
+            <Text style={styles.buttonText}>Delete</Text>
+          </TouchableOpacity>
+        </View>
+      </ScrollView>
   );
 };
 
 // Styles for the component
 const styles = StyleSheet.create({
```

Comment:
Lines 264–362:

You’ve done well with StyleSheet.create, but sprinkling raw hexes ('#f8f8f8', '#ff9800', etc.) makes palette changes harder. Think about centralizing these in a theme.ts for colors and a spacing.ts for margins/paddings (SM=8, MD=16, etc.).

Watch for “magic numbers” in margins (lines 272, 283, 289, 293, 337). A scale increases consistency.